### PR TITLE
Add all available tags to Consul Catalog Backend

### PR DESCRIPTION
**Pull request by ldez**
### What does this PR do?

- add all available tags:
  - add passTLSCert tag.
  - add HealthCheck tags.
  - add Headers tags.
  - add error pages tags.
  - add rate limit tags.
  - add whitelistSourceRange tag.
  - add frontends redirect tags.
- enhance template readability
- enhance template method readability

### Motivation

Homogenization of the providers [part2]: all providers must have the same options available.

### More

- [x] Added/updated tests
- [x] Added/updated documentation

### Additional Notes

After the review I will split `consul` package to `consul` and `consulcatalog`

Related to #618

### CODE REVIEW

File: provider/consul/consul_catalog_config_test.go [hunk 1]
```diff
@@ -0,0 +1,1253 @@
...
+func TestCatalogProviderGetInt64Attribute(t *testing.T) {
+	p := &CatalogProvider{
+		Prefix: "traefik",
+	}
+
+	testCases := []struct {
+		desc         string
+		name         string
+		tags         []string
+		defaultValue int64
```
Comment (mmatur) [inline provider/consul/consul_catalog_config_test.go]: `defaultValue` is not used in test cases

File: provider/consul/consul_catalog_config_test.go [hunk 2]
```diff
@@ -0,0 +1,1253 @@
...
+func TestCatalogProviderGetIntAttribute(t *testing.T) {
+	p := &CatalogProvider{
+		Prefix: "traefik",
+	}
+
+	testCases := []struct {
+		desc         string
+		name         string
+		tags         []string
+		defaultValue int
```
Comment (mmatur) [inline provider/consul/consul_catalog_config_test.go]: `defaultValue` is not used in test cases

File: provider/consul/consul_catalog_config_test.go [hunk 3]
```diff
@@ -0,0 +1,1253 @@
...
+func TestCatalogProviderGetBoolAttribute(t *testing.T) {
+	p := &CatalogProvider{
+		Prefix: "traefik",
+	}
+
+	testCases := []struct {
+		desc         string
+		name         string
+		tags         []string
+		defaultValue bool
```
Comment (mmatur) [inline provider/consul/consul_catalog_config_test.go]: `defaultValue` is not used in test cases

File: provider/consul/consul_catalog_config.go [hunk 4]
```diff
@@ -157,11 +188,369 @@ func (p *CatalogProvider) getSticky(tags []string) string {
...
+		return nil
+	}
+
+	return &types.MaxConn{
+		Amount:        amount,
+		ExtractorFunc: extractorFunc,
+	}
+}
+
+func (p *CatalogProvider) getHeathCheck(tags []string) *types.HealthCheck {
```
Comment (nmengin) [inline provider/consul/consul_catalog_config.go]: Do you mean `getHealthCheck`?

File: provider/consul/consul_catalog_config_test.go [hunk 5]
```diff
@@ -0,0 +1,1253 @@
...
+			t.Parallel()
+
+			result := p.getMaxConn(test.tags)
+
+			assert.Equal(t, test.expected, result)
+		})
+	}
+}
+
+func TestCatalogProviderGetHeathCheck(t *testing.T) {
```
Comment (nmengin) [inline provider/consul/consul_catalog_config_test.go]: Do you mean `TestCatalogProviderGetHealthCheck`?

File: templates/consul_catalog.tmpl [hunk 6]
```diff
@@ -1,52 +1,160 @@
...
-    amount = {{getAttribute "backend.maxconn.amount" .Attributes "" }}
-    extractorfunc = "{{getAttribute "backend.maxconn.extractorfunc" .Attributes "" }}"
+  {{ $maxConn := getMaxConn $service.Attributes }}
+  {{if $maxConn }}
+  [backends."backend-{{ getServiceBackendName $service }}".maxConn]
+    extractorFunc = "{{ $maxConn.ExtractorFunc }}"
+    amount = {{ $maxConn.Amount }}
   {{end}}
 
+  {{ $healthCheck := getHeathCheck $service.Attributes }}
```
Comment (nmengin) [inline templates/consul_catalog.tmpl]: Do you mean `getHealthCheck`?

File: provider/consul/consul_catalog_config.go [hunk 7]
```diff
@@ -16,19 +19,43 @@ import (
...
+		"hasMaxconnAttributes":    p.hasMaxConnAttributes,    // Deprecated [breaking]
+		"getSticky":               p.getSticky,               // Deprecated [breaking]
+		"hasStickinessLabel":      p.hasStickinessLabel,      // Deprecated [breaking]
+		"getStickinessCookieName": p.getStickinessCookieName, // Deprecated [breaking]
+		"getWeight":               p.getWeight,               // Deprecated [breaking] Must replaced by a simple: "getWeight": p.getFuncIntAttribute(label.SuffixWeight, 0)
+		"getProtocol":             p.getFuncStringAttribute(label.SuffixProtocol, label.DefaultProtocol),
+		"getCircuitBreaker":       p.getCircuitBreaker,
+		"getLoadBalancer":         p.getLoadBalancer,
+		"getMaxConn":              p.getMaxConn,
+		"getHeathCheck":           p.getHeathCheck,
```
Comment (nmengin) [inline provider/consul/consul_catalog_config.go]: Do you mean `getHealthCheck`?

File: provider/consul/consul_catalog_config_test.go (hunk 2 above)
Comment (ldez) [reply provider/consul/consul_catalog_config_test.go]: fixed

File: provider/consul/consul_catalog_config_test.go (hunk 1 above)
Comment (ldez) [reply provider/consul/consul_catalog_config_test.go]: fixed

File: provider/consul/consul_catalog_config_test.go (hunk 3 above)
Comment (ldez) [reply provider/consul/consul_catalog_config_test.go]: fixed

**Pull request merged**